### PR TITLE
Feature/zone_ic

### PR DESCRIPTION
**Pull request by timfelle**
- Introduction of Point zone based initial conditions.

### CODE REVIEW

File: src/fluid/flow_ic.f90 [hunk 1]
```diff
@@ -212,4 +223,37 @@ subroutine set_flow_ic_blasius(u, v, w, delta, uinf, type)
...
+    !                           zone%size)
+    !    call device_cfill_mask(v%x_d, zone_value(2), size, zone%mask_d, &
+    !                           zone%size)
+    !    call device_cfill_mask(w%x_d, zone_value(3), size, zone%mask_d, &
+    !                           zone%size)
+    ! else
+    call cfill_mask(u%x, zone_value(1), size, zone%mask, zone%size)
+    call cfill_mask(v%x, zone_value(2), size, zone%mask, zone%size)
+    call cfill_mask(w%x, zone_value(3), size, zone%mask, zone%size)
+    ! end if
```
Comment (timofeymukha) [inline src/fluid/flow_ic.f90:254]: I guess one needs the if, or?

**Review by timofeymukha — COMMENTED** [review]
Can be good to have I suppose! What are you using this for?

File: src/fluid/flow_ic.f90 [hunk 2]
```diff
@@ -173,36 +184,36 @@ subroutine set_flow_ic_blasius(u, v, w, delta, uinf, type)
...
+      case('cubic')
        bla => blasius_cubic
-    case('quartic')
+      case('quartic')
        bla => blasius_quartic
-    case('sin')
+      case('sin')
        bla => blasius_sin
-    case default
+      case default
```
[the remark this answers is not in the document]
Comment (timfelle) [reply src/fluid/flow_ic.f90:197]: We have not listed anything in the style guide about case statements.

File: src/fluid/flow_ic.f90 (hunk 1 above)
Comment (timfelle) [reply src/fluid/flow_ic.f90:254]: No, it should be removed. The `set_ic_common` always copy over the values set on the CPU side.

**Comment by timfelle** [comment]
> Can be good to have I suppose! What are you using this for?

Something like the brinkman immersed boundary methods. Initializing the flow to nothing inside can help avoid spikes in pressure on the first iterations.

**Review by njansson — APPROVED** [review]

**Review by timofeymukha — APPROVED** [review]

**Comment by timfelle** [comment]
@njansson Any idea why the Mac workflow cannot find MPI-08?

**Comment by njansson** [comment]
> @njansson Any idea why the Mac workflow cannot find MPI-08?

Not sure, but I guess it's something with the build cache

**Pull request merged**